### ISSUE TITLE
Flutter has no adaptation to Android font weights

### ISSUE THREAD

**Issue by luoyi58624** [flutter/flutter#152429]
### Use case

https://github.com/user-attachments/assets/9d7a18bc-f4ed-4cfc-8125-6335b5a385b1

### code example 

[https://github.com/luoyi58624/flutter_text_example](https://github.com/luoyi58624/flutter_text_example)

### Proposal

This difference creates a significant disconnect between flutter apps and native apps, I want fonts on flutter to sync with native Android font weights

**Comment by luoyi58624** [comment]
It's worth mentioning that about a month ago android webview also didn't follow the system font weight, but now that it's implemented, I think flutter should be able to implement it under the hood as well.
![5546875544bd3d0f747c254e01fab23](https://github.com/user-attachments/assets/f9ff3ce5-3e36-4525-845d-f60cbf75cb49)

**Comment by darshankawar** [comment]
@luoyi58624 
Please don't just post screenshots which isn't helpful to properly understand the issue.
Provide a proper description of the issue you are facing, current behavior and what expected result are you looking for. Also provide a minimal code sample along with flutter version so that we can analyze the issue properly and make actionable.

**Comment by luoyi58624** [comment]
Flutter's fonts do not follow the system, and this is not a bug, it is a feature flaw

**Comment by luoyi58624** [comment]
Android adjusts the global font size, but flutter does not change that

**Comment by darshankawar** [comment]
@luoyi58624 Check if your case resembles https://github.com/flutter/flutter/issues/139469 or not.

**Comment by luoyi58624** [comment]
Hi, I uploaded the video and sample code, in the examples, one uses the Flutter Text widget, other uses the platform view to render Android native text, when I change the global text weight, Flutter Text Widget no changes

**Comment by darshankawar** [comment]
Thanks for the video, it makes the issue clear. I think it may already be a known issue or filed, as I could see below issues which may be closely related that you can check:

https://github.com/flutter/flutter/issues/30974
https://github.com/flutter/flutter/issues/93881

**Comment by luoyi58624** [comment]
I don't know if there is any connection to the above link, I also don't know if this change in global text thickness is provided by native Android, as I only found the bold switch on the emulator, This feature may also be implemented by a third-party Rom system itself, I also find this difficult to deal with

**Comment by darshankawar** [comment]
Keeping it open for team's tracking.

**Comment by luoyi58624** [comment]
I've found that flutter3.24 renders fonts the same as browsers, And I purposely switched to 3.23.3 to verify it once, Seems to be under the influence of skia, I think this problem can be closed.
